src/kudu/tserver/simple_tablet_manager.cc

```diff
@@ -117,36 +122,208 @@ TSTabletManager::TSTabletManager(TabletServer* server)
+          JoinStrings(symm_diff, ", "));
+      return Status::InvalidArgument(msg);
+    }
+  }
+
+  RETURN_NOT_OK(SetupRaft());
+  return Status::OK();
+}
+
+Status TSTabletManager::CreateNew(FsManager *fs_manager) {
+  RaftConfigPB config;
+  if (server_->opts().IsDistributed()) {
+    RETURN_NOT_OK_PREPEND(CreateDistributedConfig(server_->opts(), &config),
+                          "Failed to create new distributed Raft config");
+  } else {
+    config.set_obsolete_local(true);
+    config.set_opid_index(consensus::kInvalidOpIdIndex);
+    RaftPeerPB* peer = config.add_peers();
+    peer->set_permanent_uuid(fs_manager->uuid());
+    peer->set_member_type(RaftPeerPB::VOTER);
+  }
+
+  RETURN_NOT_OK_PREPEND(cmeta_manager_->Create(kSysCatalogTabletId, config, consensus::kMinimumTerm),
+                        "Unable to persist consensus metadata for tablet " + kSysCatalogTabletId);
+
+  return SetupRaft();
+}
+
+Status TSTabletManager::CreateDistributedConfig(const TabletServerOptions& options,
+                                                RaftConfigPB* committed_config) {
+  DCHECK(options.IsDistributed());
+
+  RaftConfigPB new_config;
+  new_config.set_obsolete_local(false);
+  new_config.set_opid_index(consensus::kInvalidOpIdIndex);
+
+  // Build the set of followers from our server options.
+  for (const HostPort& host_port : options.tserver_addresses) {
+    RaftPeerPB peer;
+    HostPortPB peer_host_port_pb;
+    RETURN_NOT_OK(HostPortToPB(host_port, &peer_host_port_pb));
+    peer.mutable_last_known_addr()->CopyFrom(peer_host_port_pb);
+    peer.set_member_type(RaftPeerPB::VOTER);
+    new_config.add_peers()->CopyFrom(peer);
+  }
+
+  // Now resolve UUIDs.
+  // By the time a SysCatalogTable is created and initted, the masters should be
+  // starting up, so this should be fine to do.
+  DCHECK(server_->messenger());
+  RaftConfigPB resolved_config = new_config;
+  resolved_config.clear_peers();
+  for (const RaftPeerPB& peer : new_config.peers()) {
+    if (peer.has_permanent_uuid()) {
+      resolved_config.add_peers()->CopyFrom(peer);
+    } else {
+      LOG(INFO) << SecureShortDebugString(peer)
+                << " has no permanent_uuid. Determining permanent_uuid...";
+      RaftPeerPB new_peer = peer;
+      RETURN_NOT_OK_PREPEND(consensus::SetPermanentUuidForRemotePeer(server_->messenger(),
+                                                                     &new_peer),
+                            Substitute("Unable to resolve UUID for peer $0",
+                                       SecureShortDebugString(peer)));
+      resolved_config.add_peers()->CopyFrom(new_peer);
+    }
+  }
+
+  RETURN_NOT_OK(consensus::VerifyRaftConfig(resolved_config));
+  VLOG(1) << "Distributed Raft configuration: " << SecureShortDebugString(resolved_config);
+
+  *committed_config = resolved_config;
+  return Status::OK();
+}
+
+Status TSTabletManager::WaitUntilConsensusRunning(const MonoDelta& timeout) {
```

Comment:
timeout should prolly be a flag

Reply:
this timeout is the amount of timeout a single call to WaitUntilConsensusRunning is supposed to wait.
At this moment, the higher level WaitUntilRunning waits for ever or error. We will need to revisit this soon.

Reply:
left flag addition for future if necessary.